flask_restful/__init__.py

```diff
@@ -284,6 +285,17 @@ def handle_error(self, e):
         data = getattr(e, 'data', error_data(code))
         headers = {}
 
+        error_cls_name = type(e).__name__
+        if error_cls_name in self.errors:
+            custom_data = self.errors.get(error_cls_name, {})
+            code = custom_data.get('status', 500)
+            data.update(custom_data)
+
+        for typecheck, handler in self.errorhandlers:
```

Comment:
The most specialized handler should be used, not the first one matching. Otherwise, I can't catch both base classes and sub classes.